packages/eth-edge/src/services/wallets.service.ts

```diff
@@ -73,4 +73,12 @@ export class WalletsService {
             balance,
         }));
     }
+
+    async delete(address: string, userId: number): Promise<void> {
+        //Even if address is the primary key, we might want to consider going with a model
+        //where we have a primary ket composed of userId and address. This can be done in case
+        //a user decides to claim addresses that are not actually theirs.
+        this.walletRepository.delete({ userId: userId, address: address });
+        this.transactionsService.delete(address);
```

Comment:
When calling an async function here probably want to await it
```suggestion
        await this.walletRepository.delete({ userId: userId, address: address });
        await this.transactionsService.delete(address);
```